doc/release/0.19.0-notes.rst

```diff
@@ -65,6 +65,10 @@ repeatable minimizations.
 The keyword `sigma` in `scipy.optimize.curve_fit` was overloaded to also accept
 the covariance matrix of errors in the data.
 
+The method `trust-iterative` was added to the function `scipy.optimize.minimize`.
```

Comment:
My general suggestion is to maybe aim towards the name "trust-region-exact", which means that the method seeks a (near)exact solution of the trust-region subproblem. The term "iterative" is somewhat confusing as it also may refer to the trust-region algorithm as a whole.

I suspect that you (and others) may see it in a different way, but for me "trust-region-exact" sounds like a better name. If we agree on that, it should be changed throughout the whole code.